astropy/io/ascii/tests/test_read.py

```diff
@@ -543,10 +538,10 @@ def test_fill_values(fast_reader):
     data = ascii.read(
         f, fill_values=("a", "1"), fast_reader=fast_reader, **testfile["opts"]
     )
-    assert_true((data["a"].mask == [False, True]).all())
-    assert_true((data["a"] == [1, 1]).all())
-    assert_true((data["b"].mask == [False, True]).all())
-    assert_true((data["b"] == [2, 1]).all())
+    assert (data["a"].mask == [False, True]).all()
+    assert (data["a"] == [1, 1]).all()
+    assert (data["b"].mask == [False, True]).all()
+    assert (data["b"] == [2, 1]).all()
```

Comment:
```suggestion
    assert_equal(data["a"].mask, [False, True])
    assert_equal(data["a"], [1, 1])
    assert_equal(data["b"].mask, [False, True])
    assert_equal(data["b"], [2, 1])
```
I agree the `numpy.testing` function provides rather better diagnostics here even though the arrays are small.
This applies to all the `assert (...).all()` checks _here and below_.
In contrast, all the `assert_equal` uses _above this line_ (basically all comparisons of some `dtype.names` are comparing tuples of non-numeric objects and should use `assert ==` – not just those in the suggestions.